demisto_sdk/commands/common/hook_validations/pack_unique_files.py

```diff
@@ -798,3 +803,29 @@ def validate_core_pack_dependencies(self, dependencies_packs):
             if self._add_error((error_message, error_code), file_path=self.pack_path):
                 return False
         return True
+
+    @error_codes('PA132')
+    def should_pack_be_hidden(self) -> bool:
+        """
+        Validates whether a pack should be hidden according to the following rules:
+
+        1. if the pack is not already hidden.
+        2. if all the content items (playbooks/scripts/integrations) are deprecated.
+
+        Returns:
+            bool: True if pack should be hidden, False if it shouldn't.
+        """
+        deprecated_pack_content_items = DeprecatedPackContentItems(self.pack_path)
+        if deprecated_pack_content_items.should_pack_be_hidden():
+            error_message, error_code = Errors.pack_should_be_hidden(
+                pack_name=self.pack,
+                deprecated_content_items=deprecated_pack_content_items.get_deprecated_content_items_report()
+            )
+            if self._add_error(
+                (error_message, error_code),
+                file_path=self.pack_meta_file,
+                should_print=True,
+                suggested_fix=Errors.suggest_fix(file_path=os.path.join(self.pack_path, self.pack_meta_file))
+            ):
+                return True
+        return False
```

Comment:
The ```Return False``` is not inside the ```if deprecated_pack_content_items.should_pack_be_hidden()``` statement. so it can be something like this:

```
deprecated_pack_content_items = DeprecatedPackContentItems(self.pack_path)
        if deprecated_pack_content_items.should_pack_be_hidden() is True:
            error_message, error_code = Errors.pack_should_be_hidden(self.pack)
            return self._add_error(
                (error_message, error_code),
                file_path=self.pack_meta_file,
                should_print=True,
                suggested_fix=Errors.suggest_fix(file_path=os.path.join(self.pack_path, self.pack_meta_file))
            )
        return False
```